### PR TITLE
feat(3d-tiles): some improvements

### PR DESCRIPTION
**Pull request by dariaterekhovaae**
Fix for review comments from https://github.com/visgl/loaders.gl/pull/2578

### CODE REVIEW

File: modules/3d-tiles/src/tz3-loader.ts [hunk 1]
```diff
@@ -34,8 +38,10 @@ export const TZ3Loader: LoaderWithParser<Buffer, never, TZ3LoaderOptions> = {
  * @param options options
  * @returns requested file
  */
-async function parse3tz(data: ArrayBuffer, options: TZ3LoaderOptions = {}) {
-  return (await parse3tzFromProvider(new DataViewFile(new DataView(data)))).getFile(
-    options.tz3?.path ?? ''
-  );
+async function parse3tz(
```
Comment (ibgreen) [inline modules/3d-tiles/src/tz3-loader.ts:41]: `parse3DTilesArchive`

**Review by ibgreen — APPROVED** [review]

File: modules/3d-tiles/src/index.ts [hunk 2]
```diff
@@ -2,8 +2,8 @@
 export {Tiles3DLoader} from './tiles-3d-loader';
 export {CesiumIonLoader} from './cesium-ion-loader';
 export {Tile3DSubtreeLoader} from './tile-3d-subtree-loader';
-export type {TZ3LoaderOptions} from './tz3-loader';
-export {TZ3Loader} from './tz3-loader';
+export type {Tiles3DArchiveFileLoaderOptions} from './tz3-loader';
```
Comment (belom88) [inline modules/3d-tiles/src/index.ts:5]: rename files and folders: `tz3**` -> `3d-tiles-archive**`. We can start file name from a number

File: modules/3d-tiles/src/tz3/tz3-archive.ts [hunk 3]
```diff
@@ -9,24 +9,27 @@ type CompressionHandler = (compressedFile: ArrayBuffer) => Promise<ArrayBuffer>;
...
+  /** FileProvider with whe whole file */
+  private zipArchiveFile: FileProvider;
+  /** hash info */
   private hashArray: HashElement[];
-  constructor(tz3Archive: FileProvider, hashFile: HashElement[]) {
-    this.tz3Archive = tz3Archive;
+
+  /**
+   * creates Tiles3DArchive handler
+   * @param zipArchiveFile - FileProvider with whe whole file
```
Comment (belom88) [inline modules/3d-tiles/src/tz3/tz3-archive.ts:28]:
```suggestion
   * @param zipArchiveFile - FileProvider with the whole file
```

File: modules/3d-tiles/src/tz3/tz3-archive.ts [hunk 4]
```diff
@@ -47,7 +50,7 @@ export class TZ3Archive {
     }
     const decompressedFile = Buffer.from(data);
 
-    return decompressedFile;
+    return decompressedFile.buffer;
```
Comment (belom88) [inline modules/3d-tiles/src/tz3/tz3-archive.ts:53]: `this.getFileBytes` returns `ArrayBuffer`. If you want to handle `null`, it is a weird way. I'd recomment `throw` in null case. That means that the file has not been found

File: modules/3d-tiles/src/tz3/tz3-archive.ts [hunk 5]
```diff
@@ -9,24 +9,27 @@ type CompressionHandler = (compressedFile: ArrayBuffer) => Promise<ArrayBuffer>;
...
 };
 
 /**
  * Class for handling information about 3tz file
  */
-export class TZ3Archive {
-  private tz3Archive: FileProvider;
+export class Tiles3DArchive {
+  /** FileProvider with whe whole file */
+  private zipArchiveFile: FileProvider;
```
Comment (belom88) [inline modules/3d-tiles/src/tz3/tz3-archive.ts:22]:
```suggestion
  private fileProvider: FileProvider;
```

**Review by belom88 — CHANGES_REQUESTED** [review]

**Review by belom88 — APPROVED** [review]

**Pull request merged**